### DISCUSSION TITLE
How to tell Yarn to cache assets for a different architecture? (I need Linux and Darwin ones)

### DISCUSSION THREAD

**Discussion by FezVrasta** [yarnpkg/berry#3872] [Q&A] [answered]
Hi, I'm using the zero-installs approach and I'm having a problem with the esbuild package, when I install it, only the Darwin binaries are added to the cache, but I need also the Linux ones so that my CI can run with `--immutable-cache`. What's the correct way to achieve this?

The only solution I found, so far, is to add the different binaries as direct dependencies of my project. I don't know how correct this is though.

```
    "esbuild-darwin-64": "0.13.15",
    "esbuild-darwin-arm64": "0.13.15",
    "esbuild-linux-64": "0.13.15",
```

**Comment by merceyz** [comment, marked as answer]
To quote the changelog:
> Yarn won't install anymore packages that don't match the current OS. Should you need to (for example if you check-in your cache), use the [supportedArchitectures](https://yarnpkg.com/configuration/yarnrc#supportedArchitectures) field to manually enforce the presence of specific architectures.
> https://github.com/yarnpkg/berry/blob/f61bf6fb26d2211ac187dc4b37d68f00be4aee9f/CHANGELOG.md#installs-1

**Reply by FezVrasta** [reply]
Ah thanks! I couldn't find it 🤦‍♂️
